src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1556,8 +1557,22 @@ def aks_create(cmd, client, resource_group_name, name, ssh_key_value,  # pylint:
     retry_exception = Exception(None)
     for _ in range(0, max_retry):
         try:
-            return sdk_no_wait(no_wait, client.create_or_update,
+           result = sdk_no_wait(no_wait, client.create_or_update,
                                resource_group_name=resource_group_name, resource_name=name, parameters=mc)
+           # add cluster spn with Monitoring Metrics Publisher role assignment to the cluster resource
+           if monitoring:
+            from msrestazure.tools import resource_id
+            cluster_resource_id = resource_id(
+                    subscription=subscription_id,
+                    resource_group=resource_group_name,
+                    namespace='Microsoft.ContainerService', type='managedClusters',
```

Comment:
define constants?

Reply:
CLI code follows the inline constants. I followed the same convention for namespace and type.